src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -312,7 +312,7 @@ def load_arguments(self, _):
         c.argument('enable_vmss', action='store_true', help='To be deprecated. Use vm_set_type instead.', deprecate_info=c.deprecate(redirect='--vm-set-type', hide=True))
         c.argument('zones', zones_type, options_list=['--zones', '-z'], help='Space-separated list of availability zones where agent nodes will be placed.')
         c.argument('ppg')
-        c.argument('enable_encryption_at_host', arg_type=get_three_state_flag(), help='Enable EncryptionAtHost.')
+        c.argument('enable_encryption_at_host', action='store_true')
```

Comment:
why remove the help message?

Reply:
The declaration of this option in azure-cli/acs is an ordinary bool flag, and this option does not support update. So remove the confusing three state flag.

Reply:
I'd like to confirm whether changing `get_three_state_flag()` to `action='store_true'` will cause a breaking change to users? Did any users use `--enable-encryption-at-host False` before?

Reply:
Hi @zhoxing-ms, I think this wouldn't be a breaking change to users.

This property does not support being updated, it is only allowed to be used when creating an aks cluster, and the default value is false, only need to be specified when this feature needs to be enabled.

In the [official doc](https://learn.microsoft.com/en-us/azure/aks/enable-host-encryption#use-host-based-encryption-on-new-clusters), the example given in the guide is in the form of `az aks create xxx --enable-encryption-at-host`.

Besides the option is also supported in azure-cli/acs, and the [same parameter](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/_params.py#L259) declared there is a normal bool flag.